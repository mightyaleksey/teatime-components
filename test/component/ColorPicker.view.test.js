'use strict';

const ColorPicker = require('../../component/ColorPicker');
const React = require('react');
const renderer = require('react-test-renderer');

test('defaultProp #size should have a value `m`', () => {
  expect(ColorPicker.defaultProps.size).toBe('m');
});

test('Renders correctly with default props', () => {
  const tree = renderer.create(
    <ColorPicker
      name='control'/>
  );

  expect(tree).toMatchSnapshot();
});

test('Renders correctly with defined props', () => {
  const tree = renderer.create(
    <ColorPicker
      autoFocus
      className='outer'
      disabled
      id='controls-id'
      name='control'
      placeholder='rrggbb'
      value='#bebebe'/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
