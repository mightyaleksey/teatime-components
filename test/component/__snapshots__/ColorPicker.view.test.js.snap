// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with default props 1`] = `
<span
  className="container"
  onChange={undefined}
>
  <button
    className="button"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "",
      }
    }
  />
  <input
    autoFocus={undefined}
    className="control"
    disabled={undefined}
    id={undefined}
    name="control"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder={undefined}
    type="text"
    value=""
  />
  <span
    className="menu isClosedMenu"
  />
</span>
`;

exports[`Renders correctly with defined props 1`] = `
<span
  className="container outer"
  onChange={undefined}
>
  <button
    className="button"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#bebebe",
      }
    }
  />
  <input
    autoFocus={true}
    className="control"
    disabled={true}
    id="controls-id"
    name="control"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="rrggbb"
    type="text"
    value="BEBEBE"
  />
  <span
    className="menu isClosedMenu"
  />
</span>
`;
